### ISSUE TITLE
Git plugin freaks out when there's a conflict

### ISSUE THREAD

**Issue by ffxsam** [robbyrussell/oh-my-zsh#4932]
I did a `git rebase master` which resulted in a conflict. And now every time I hit enter at the zsh prompt, I get this:

    VCS_INFO_get_data_git:223: no such file or directory: .git/rebase-apply/msg-clean

**Comment by mcornella** [comment]
This is a bug in zsh's codebase, fixed in https://github.com/zsh-users/zsh/commit/7a1678767f44f200295d88dfea94902718275266. If you can, update your zsh to 5.2.

**Comment by ffxsam** [comment]
How do I tell which version I currently have? And how do I update to a specific version?

**Comment by mcornella** [comment]
`zsh --version` gets you the version of zsh. Updating it depends on whatever your operating system is.

**Comment by ffxsam** [comment]
Oh, I was thinking oh-my-zsh, not zsh. Got it. I'm using zsh 5.2.

**Comment by auszone** [comment]
+1

**Comment by mcornella** [comment]
This should be in the latest version of zsh 5.2. Perhaps your platform hasn't shipped a modified version of 5.2. For instance, I'm using Debian testing and my installed zsh package is version `5.2-5`, meaning it's the 5th or 6th modified version of zsh 5.2. What system are you using?

**Comment by ffxsam** [comment]
Mac OS 10.11.5. zsh 5.2 (x86_64-apple-darwin15.4.0). Installed via Homebrew.

**Comment by mcornella** [comment]
That's weird, the latest Homebrew formula was [updated on April 9](https://github.com/Homebrew/homebrew-core/blob/80af652a7693937a2802988eccaa51f1b6b3af29/Formula/zsh.rb) and the [`tar.gz` from sourceforge](https://sourceforge.net/projects/zsh/files/zsh/5.2/) was last updated on December 2015, both long after the fix was introduced to the 5.2 version of zsh. Perhaps you can `brew update zsh`, or uninstall the install back again?

**Comment by SimenB** [comment]
I'm having the same problem, installing and reinstalling from homebrew doesn't do anything (it's the same version)

**Comment by jtheoof** [comment]
@SimenB looks like your `$PATH` is off.

```
$ echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

If you see `/usr/bin` first then you will not see the one installed by Homebrew. See the [doc](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/FAQ.md#my-mac-apps-dont-find-usrlocalbin-utilities).

**Comment by jtheoof** [comment]
Installing the latest zsh (`zsh 5.2 (x86_64-apple-darwin15.4.0)`) did not fix the error for me.

**Comment by SimenB** [comment]
Relevant parts of my `$PATH` is `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin`, so I should be good (only `nvm` and `rvm` before it)

**Comment by WayneYe** [comment]
+1 for @jtheoof 

I installed latest zsh 5.2 and updated my /etc/shells, i am using **OSX El Capitan v10.11.6**, below is my environment:

```
>  which zsh
/usr/local/bin/zsh
>  echo $SHELL
/usr/local/bin/zsh
>  zsh --version
zsh 5.2 (x86_64-apple-darwin15.4.0)
>  git --version
git version 2.9.2
```

What I got:
```
> git status
rebase in progress; onto a752d68
You are currently rebasing branch 'prod' on 'a752d68'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

	both modified:   .travis.yml

no changes added to commit (use "git add" and/or "git commit -a")
VCS_INFO_get_data_git:223: no such file or directory: .git/rebase-apply/msg-clean
```

**Comment by js3692** [comment]
+1

**Comment by esgott** [comment]
+1

**Comment by mcornella** [comment]
To everyone that has this problem, please run the following and post the output:

```zsh
zsh --version
for d ($fpath); do
    f="$d/VCS_INFO_get_data_git"
    if [[ -f "$f" ]]; then
        command shasum "$f"
    fi
done
```

**Comment by esgott** [comment]
```
zsh 5.2 (x86_64-apple-darwin15.0.0)

123fbb5938bd16a8aa0c92a5df9ede62ead06dd1  /usr/share/zsh/5.0.8/functions/VCS_INFO_get_data_git
```
I think this explains the behaviour. Looking at my `fpath` variable, there are multiple elements for the old version on OSX

**Comment by mcornella** [comment]
Wait... There isn't a 5.2.0 `VCS_INFO_...` file?

**Comment by esgott** [comment]
@mcornella There is, but the zsh `functions` directory from 5.2 was not in the `fpath` variable. I think it is a Homwbrew package bug. Do you agree?

**Comment by mcornella** [comment]
I'm not sure actually. Try running `zsh -f` and looking at what `$fpath` is.

**Comment by esgott** [comment]
This is my fpath,  sourceing oh-my-zsh from my `.zshrc` is commented out

```
 $ echo $fpath
/usr/local/share/zsh/site-functions /usr/share/zsh/site-functions /usr/share/zsh/5.0.8/functions
```

**Comment by js3692** [comment]
@mcornella:

This is the output:
`zsh 5.2 (x86_64-apple-darwin15.4.0)`
`123fbb5938bd16a8aa0c92a5df9ede62ead06dd1  /usr/share/zsh/5.0.8/functions/VCS_INFO_get_data_git`

And `$fpath`:
`/usr/local/share/zsh/site-functions /usr/local/Cellar/zsh/5.2/share/zsh/functions`

Any ideas...?

**Comment by mcornella** [comment]
@apjanke is it possible that there is an issue with Homebrew here? Can you try to reproduce this as well? 

BTW I never got to congratulate you about your promotion as official collaborator at Homebrew so... congrats! 👏

**Comment by SimenB** [comment]
`zsh --version` => `zsh 5.2 (x86_64-apple-darwin15.4.0)`

The loop => `123fbb5938bd16a8aa0c92a5df9ede62ead06dd1  /usr/share/zsh/5.0.8/functions/VCS_INFO_get_data_git`

`$fpath` without `oh-my-zsh` stuff => `/usr/local/share/zsh/site-functions /usr/share/zsh/site-functions /usr/share/zsh/5.0.8/functions`

**Comment by jtheoof** [comment]
```
$ zsh --version
zsh 5.0.8 (x86_64-apple-darwin15.0)
```

```
$ for d ($fpath); do
    f="$d/VCS_INFO_get_data_git"
    if [[ -f "$f" ]]; then
        command shasum "$f"
    fi
done
123fbb5938bd16a8aa0c92a5df9ede62ead06dd1  /usr/share/zsh/5.0.8/functions/VCS_INFO_get_data_git
```

```
$ cat -n /usr/share/zsh/5.0.8/functions/VCS_INFO_get_data_git | ag '^\s*223'
   223 	    subject="${$(< "${patchdir}/msg-clean")[(f)1]}"
```

**Comment by apjanke** [comment]
TL;DR: Homebrew zsh 5.2 has the fix; it's your `chsh` and not your `$PATH` that probably matters.

> @apjanke is it possible that there is an issue with Homebrew here? Can you try to reproduce this as well? 

I can reproduce this on my OS X 10.11.6 machine, but only with the system-supplied zsh 5.0.8. I have an updated Homebrew-installed zsh 5.2, and it is not affected. So it looks like the Homebrew version includes the fix. To be affected, you also need to be using a theme that uses zsh's supplied `vcs_info` instead of OMZ's own git theming support, such as the `pure` theme.

```
openssl git/man-page-formatting*
❯ pwd
/Users/janke/local/repos/openssl/openssl
VCS_INFO_get_data_git:223: no such file or directory: .git/rebase-apply/msg-clean

openssl git/man-page-formatting*
❯ echo $ZSH_VERSION
5.0.8
VCS_INFO_get_data_git:223: no such file or directory: .git/rebase-apply/msg-clean

openssl git/man-page-formatting*
❯ zsh

openssl git/man-page-formatting*
❯ pwd
/Users/janke/local/repos/openssl/openssl

openssl git/man-page-formatting*
❯ echo $ZSH_VERSION
5.2

openssl git/man-page-formatting*
❯
```

There's a possible interaction with Homebrew, insofar as you may not be running the Homebrew-installed `zsh` in new terminal sessions, regardless of what `which zsh` gives you inside a session. This means that `zsh --version` is not necessarily the one you're running. The first `zsh` in a new terminal session is controlled by your `chsh` setting. Use `echo $ZSH_VERSION` to see what version the current session is running. 


For example, I'm on OS X 10.11.6, and I have `zsh` 5.2 installed via Homebrew. But I still get the system `zsh` 5.0.8 by default, since I have `/bin/zsh` as my default shell.

```
$ which zsh
/usr/local/bin/zsh
[~]
$ zsh --version
zsh 5.2 (x86_64-apple-darwin13.4.0)
[~]
$ echo $ZSH_VERSION
5.0.8
```

If that's your issue, and you want to use the newer 5.2 zsh by default, do a `chsh -s $(homebrew --prefix)/bin/zsh` (after making sure it's in your `/etc/shells` file). @WayneYe, I suspect this is your situation, since you mentioned updating `/etc/shells` but not running a `chsh`. I'd guess this is true for others here, since you're checking the version with `zsh --version` and not `$ZSH_VERSION`.

Or, your terminal emulator might be set up to run a specific `zsh` as your login shell, and it could be pointing to `/bin/zsh` instead of `/usr/local/bin/zsh`. If that's the case, I'd just switch your terminal profile to using your default login shell, so it respects your `chsh` settings.

And thank you Marc!

**Comment by mcornella** [comment]
Ok everyone please check you're running 5.2.0 with `echo $ZSH_VERSION`.

**Comment by WayneYe** [comment]
@apjanke you are super! your solution fixed this problem!!

**Comment by ffxsam** [comment]
@apjanke And all this time I was using the wrong binary (`/bin/zsh`)! Thanks for this!

**Comment by mcornella** [comment]
Ok I'll close this issue for now. If someone else can't solve it I'll reopen. Thanks Andrew!

**Comment by yordis** [comment]
Just a little feedback from my problem.

I didn't have `echo $ZSH_VERSION` 5.2 instead I had `5.0`. I installed the latest version from Brew but I had to add `/usr/local/bin/zsh` to `/etc/shells` and then everything was working fine.

**Comment by thitami** [comment]
I came across this issue today. I am on a MAC OSX El Capitan and this is the trace:


```

 $ brew install --without-etcdir zsh
Updating Homebrew...
Warning: zsh-5.3.1 already installed

dscl . -read /Users/$USER UserShell
UserShell: /usr/local/bin/zsh
$ which zsh
/bin/zsh
 $ zsh --version
 zsh 5.0.8 (x86_64-apple-darwin15.0)

chsh -s /usr/local/bin/zsh
Changing shell for thitami.
Password for thitami:
chsh: no changes made
```

``` $ sudo vim /etc/shells
[....]
/bin/zsh
/bin/csh/bin/bash```

Any thoughts ?

**Comment by kunkun-tang** [comment]
+1

**Comment by edouard-lopez** [comment]
On Linux Mint/Ubuntu U got:

```
zsh 5.1.1 (x86_64-ubuntu-linux-gnu)
60bec9ba8aa6f76ce605239846211159a5a20f1c  /usr/share/zsh/functions/VCS_Info/Backends/VCS_INFO_get_data_git
VCS_INFO_get_data_git:225: no such file or directory: /data/projects/coaxis-opt/.git/rebase-apply/msg-clean
```

Is there a PPA with ZSH for ubuntu to upgrade to `5.2` ?

**Comment by danedmunds** [comment]
Just to help people more...
Using mcornella's tip (`echo $ZSH_VERSION`) I was able to figure out that I was actually running an old zsh version but `chsh -s /usr/local/bin/zsh` wasn't working for me. This finally did it for me: https://stackoverflow.com/questions/17648621/how-do-i-update-zsh-to-the-latest-version
